File: Java programming/problems on recursion/problem11.java
/*
11. Write a recursive program which accept number from user and return largest digit 
	
	Input : 87983 
	
	Output : 9  
	
	Author :Milind Nalawade
	
	Date : 26/08/2021
	
	Language : Java 14.0.2

*/

import java.lang.*;
import java.util.*;

class Logic
{
	public static int max=0,mod=0;
	static void Mul(int no)
	{
		if(no==0)
		{
			return ;
		}
		else
		{
			mod=no%10;
			if(max<mod)
			{
				max=mod;
			}
			no=no/10;
			Mul(no);
		}
	}	
}

class problem11 extends Logic
{
	public static	void main(String argv[])
	{
		Scanner sobj=new Scanner(System.in);
		System.out.print("\n Input : ");
		int no=sobj.nextInt();
		Mul(no);
		System.out.print("\n Output : "+max+"\n");
	}
}

File: C++ Programming/problems on recursion/problem11.cpp
/*
11. Write a recursive program which accept number from user and return largest digit 
	
	Input : 87983 
	
	Output : 9  
	
	Author :Milind Nalawade
	
	Date : 26/08/2021
	
	Language : C++ programming
*/
using namespace std;
#include<iostream>
int Max=0;
int Output(int no)
{
    int mod=0;
    if(no==0)
    {
        return Max;
    }
    else
    {
        mod=no%10;
        if(Max<mod)
        {
            Max=mod;
        }
        no=no/10;
        Output(no);
    }
}

int main()
{
    int no=0,ret=0;
    cout<<"\n Input : ";
    cin>>no;
    ret=Output(no);
    cout<<"\n Output : "<<ret;
}